src/Umbraco.SampleSite/Controllers/LessonsController.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Umbraco.Core;
+using Umbraco.Core.Cache;
+using Umbraco.Core.Configuration;
+using Umbraco.Core.Logging;
+using Umbraco.Web.Editors;
+using Umbraco.Web.Mvc;
+using Umbraco.Web.WebApi.Filters;
+
+namespace Umbraco.SampleSite.Controllers
+{
+    [PluginController("Starterkit")]
+    public class LessonsController : UmbracoAuthorizedJsonController
+    {
+        /// <summary>
+        /// Fetches available lessons for a given section from our.umbaco.org
+        /// </summary>
+        /// <param name="path">Name of the documentation section to fetch from, ex: "getting-started", "Tutorials/Starter-kit/Lessons" </param>
+        /// <returns></returns>
+        [ValidateAngularAntiForgeryToken]
+        public async Task<IEnumerable<Lesson>> GetLessons(string path)
+        {
+            //information for the request, so we could in the future filter by user, allowed sections, langugae and user-type
+            var user = Security.CurrentUser;
+            var userType = user.UserType.Alias;
+            var allowedSections = string.Join(",", user.AllowedSections);
+            var language = user.Language;
+            var version = UmbracoVersion.GetSemanticVersion().ToSemanticString();
+
+            //construct the url and cache key
+            var url = string.Format("https://our.umbraco.org/Umbraco/Documentation/Lessons/GetDocsForPath?path={0}&userType={1}&allowedSections={2}&lang={3}&version={4}", path, userType, allowedSections, language, version);
+            var key = "umbraco-lessons-" + userType + language + allowedSections.Replace(",", "-") + path;
+
+            var result = new List<Lesson>();
+
+            Func<List<Lesson>> fetchLesson = () =>
```

Comment:
We don't have to change this but generally it's nicer to just put this callback inline, then you don't have to declare the Func<List<Lesson>>, you just use the lamba like:

```cs
result = ApplicationContext.ApplicationCache.RuntimeCache.GetCacheItem<List<Lesson>>(
    key, 
    () => { //TODO Your code goes here }, 
    new TimeSpan(0, 30, 0));
```